bundle/compliance/policy/aws_iam/ensure_hardware_mfa.rego

```diff
@@ -0,0 +1,11 @@
+package compliance.policy.aws_iam.ensure_hardware_mfa
+
+import data.compliance.policy.aws_iam.data_adapter
+
+default ensure_hardware_device = false
+
+ensure_hardware_device {
+	data_adapter.iam_user.mfa_active
+	device := data_adapter.iam_user.mfa_devices[_]
+	not device.is_virtual
```

Comment:
this will pass if one of the MFA devices is hardware, and the benchmarks say all of the MFA devices for the root account should be hardware as I understand it.

From the benchmark audit:

> If the output contains one MFA with the following Serial Number, it means the MFA is
virtual, not hardware and the account is not compliant with this recommendation:

Reply:
Actually, I'm not sure about it.
One can use a hardware MFA device as a primary authentication method, but also have a software (virtual) MFA device as a backup.
Trivy implemented it the same way - [trivy implementation](https://github.com/aquasecurity/defsec/blob/4f3187c78d6b00afbfc741c7d8c46c663774fc25/internal/rules/aws/iam/enforce_root_hardware_mfa.go)

Reply:
A comment has been added.